packages/editor/src/bindings/post-meta.js

```diff
@@ -9,21 +9,75 @@ import { store as coreDataStore } from '@wordpress/core-data';
 import { store as editorStore } from '../store';
 import { unlock } from '../lock-unlock';
 
+function getMetadata( registry, context ) {
+	let metaFields = {};
+	const {
+		type,
+		is_custom: isCustom,
+		slug,
+	} = registry.select( editorStore ).getCurrentPost();
+	const { getPostTypes, getEditedEntityRecord } =
+		registry.select( coreDataStore );
+
+	const { getRegisteredPostMeta } = unlock(
+		registry.select( coreDataStore )
+	);
+
+	// Inherit the postType from the slug if it is a template.
+	if ( ! context?.postType && type === 'wp_template' ) {
+		// Get the 'kind' from the start of the slug.
+		// Use 'post' as the default.
+		let postType = 'post';
+		const isGlobalTemplate = isCustom || slug === 'index';
+		if ( ! isGlobalTemplate ) {
+			const [ kind ] = slug.split( '-' );
+			if ( kind === 'page' ) {
+				postType = 'page';
+			} else if ( kind === 'single' ) {
+				const postTypes =
+					getPostTypes( { per_page: -1 } )?.map(
+						( entity ) => entity.slug
+					) || [];
+
+				// Infer the post type from the slug.
+				// TODO: Review, as it may not have a post type. http://localhost:8888/wp-admin/site-editor.php?canvas=edit
+				const match = slug.match(
+					`^single-(${ postTypes.join( '|' ) })(?:-.+)?$`
+				);
+				postType = match ? match[ 1 ] : 'post';
+			}
+		}
+		const fields = getRegisteredPostMeta( postType );
+
+		// Populate the `metaFields` object with the default values.
+		Object.entries( fields || {} ).forEach( ( [ key, props ] ) => {
+			// If the template is global, skip the fields with a subtype.
+			if ( isGlobalTemplate && props.object_subtype ) {
+				return;
+			}
+			metaFields[ key ] = props.default;
```

Comment:
What will happen when the `default` is undefined or empty?

Reply:
It says:
`Apart from that, when no default is provided, it should still show the meta key.`

I'll test it.

Reply:
Registering a field without default value:
<img width="313" alt="Screenshot 2024-09-09 at 22 30 37" src="https://github.com/user-attachments/assets/e0797214-3c84-4a85-933f-4b1ba0179ba1">

is returning the meta key:
<img width="833" alt="Screenshot 2024-09-09 at 22 30 25" src="https://github.com/user-attachments/assets/f74376bc-f193-4b40-b484-e03bd9595dc2">

Reply:
It seems that is returning an empty string, as documented here:
https://developer.wordpress.org/reference/functions/get_metadata_default/#description

<img width="1507" alt="Screenshot 2024-09-09 at 22 44 25" src="https://github.com/user-attachments/assets/a1d176e8-6b73-446e-9f67-d0c1fdd332db">

Reply:
If `single` is `false` in `register_meta()` args. `get_meta_data_default()` should return an empty array.

<img width="783" alt="Screenshot 2024-09-09 at 22 56 21" src="https://github.com/user-attachments/assets/a559508b-0662-4e36-b65b-cc8e43454eec">

When testing, it returned undefined. But still is showing just the key. I should review if we are covering all the cases with tests.

<img width="1512" alt="Screenshot 2024-09-09 at 22 55 45" src="https://github.com/user-attachments/assets/579c1492-d853-4257-83b9-e15a4afeff72">